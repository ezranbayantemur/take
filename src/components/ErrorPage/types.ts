export interface ErrorPageProps {
  message?: string;
  onRetry: () => void;
}
