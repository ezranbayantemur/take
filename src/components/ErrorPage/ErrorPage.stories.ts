import ErrorPage from './ErrorPage';
import type {Meta, StoryObj} from '@storybook/react-native';
import type {ErrorPageProps} from './types';

const ErrorPageMeta: Meta<ErrorPageProps> = {
  title: 'ErrorPage',
  component: ErrorPage,
};

export default ErrorPageMeta;

export const Basic: StoryObj<ErrorPageProps> = {};
