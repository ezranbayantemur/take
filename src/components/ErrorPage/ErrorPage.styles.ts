import {color, font, padding, typography} from '@style';
import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  message_text: {
    fontFamily: font.REGULAR,
    fontSize: typography.BIG,
    marginVertical: padding.MEDIUM,
    color: color.SMOKE,
  },
});
