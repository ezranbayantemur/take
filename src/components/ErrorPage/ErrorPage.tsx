import React from 'react';
import {SafeAreaView, Text} from 'react-native';
import Button from '../Button';
import type {ErrorPageProps} from './types';

import styles from './ErrorPage.styles';

const ErrorPage = ({
  message = 'Bir problem oluÅŸtu',
  onRetry,
}: ErrorPageProps) => {
  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.message_text}>{message}</Text>
      <Button text="Tekrar Dene" onPress={onRetry} />
    </SafeAreaView>
  );
};

export default ErrorPage;
