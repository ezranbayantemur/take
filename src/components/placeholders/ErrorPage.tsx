import {Text, StyleSheet, View} from 'react-native';
import React from 'react';
import Icon from '../Icon';
import {color, font, padding, typography} from '@style';

const ErrorPage = () => {
  return (
    <View style={styles.container}>
      <Text style={styles.logo}>take</Text>
      <Icon name="cross" style={styles.icon} />
      <Text style={styles.text}>Bir problem ile olu≈ütu</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignSelf: 'center',
  },
  logo: {
    margin: padding.LARGE,
    textAlign: 'center',
    color: color.MONZA,
    fontFamily: font.BOLD_ITALIC,
    fontSize: typography.HUGE,
  },
  text: {
    marginTop: padding.MEDIUM,
    color: color.SMOKE,
    fontFamily: font.BOLD_ITALIC,
    fontSize: typography.BIG,
  },
  icon: {
    textAlign: 'center',
    color: color.SMOKE,
    fontSize: typography.HUGE,
  },
});

export default ErrorPage;
