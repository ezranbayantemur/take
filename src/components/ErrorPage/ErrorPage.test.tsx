import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import ErrorPage from './ErrorPage';

const mockOnRetry = jest.fn();

let wrapper: ReturnType<typeof render>;

describe('ErrorPage unit tests', () => {
  beforeEach(() => {
    wrapper = render(
      <ErrorPage message="test error message" onRetry={mockOnRetry} />,
    );
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render default message correctly', () => {
    wrapper.rerender(<ErrorPage onRetry={mockOnRetry} />);
    expect(wrapper.queryByText('Bir problem oluÅŸtu')).not.toBeNull();
  });

  it('should render custom message correctly', () => {
    expect(wrapper.queryByText('test error message')).not.toBeNull();
  });

  it('should trigger onRetry correctly', () => {
    const retry = wrapper.getByText('Tekrar Dene');

    fireEvent(retry, 'onPress');

    expect(mockOnRetry).toBeCalled();
  });
});
