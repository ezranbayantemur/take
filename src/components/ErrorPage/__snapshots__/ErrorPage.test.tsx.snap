// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage unit tests should render correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#BDBDBD",
        "fontFamily": "Lato-Regular",
        "fontSize": 20,
        "marginVertical": 16,
      }
    }
  >
    test error message
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#DB040E",
        "borderRadius": 20,
        "margin": 8,
        "opacity": 1,
        "padding": 16,
      }
    }
    testID="undefined_button_touchable"
  >
    <Text
      style={
        {
          "color": "#FBFBFB",
          "fontFamily": "Lato-Bold",
          "fontSize": 15,
        }
      }
      testID="undefined_button_text"
    >
      Tekrar Dene
    </Text>
  </View>
</RCTSafeAreaView>
`;
